subsys/net/lib/tls_credentials/tls_credentials_shell.c

```diff
@@ -0,0 +1,813 @@
+			"Non-printable characters were included in the output and filtered. "
+			"Have you selected the correct storage format?\n"
+		);
+	}
+
+cleanup:
+	/* Clear buffers when done. */
+	memset(cred_out_buf, 0, sizeof(cred_out_buf));
+	shell_clear_cred_buf(sh);
+	return err;
+}
+
+/* Lists credentials in credential store. */
+static int tls_cred_cmd_list(const struct shell *sh, size_t argc, char *argv[])
+{
+	int err = 0;
+	size_t digest_size;
+	sec_tag_t sectag = TLS_SEC_TAG_NONE;
+	struct tls_credential *cred;
+	int count = 0;
+
+	sec_tag_t sectag_filter = TLS_SEC_TAG_NONE;
+	enum tls_credential_type type_filter = TLS_CREDENTIAL_NONE;
+
+	/* Lock credentials so that we can safely use internal access functions. */
+	credentials_lock();
+
+	/* Sectag filter was provided, parse it. */
+	if (argc >= 2) {
+		err = shell_parse_cred_sectag(sh, argv[1], &sectag_filter, true);
+		if (err) {
+			goto cleanup;
+		}
+	}
+
+	/* Credential type filter was provided, parse it. */
+	if (argc >= 3) {
+		err = shell_parse_cred_type(sh, argv[2], &type_filter, true);
+		if (err) {
+			goto cleanup;
+		}
+	}
+
+	/* Scan through all occupied sectags */
+	while ((sectag = credential_next_tag_get(sectag)) != TLS_SEC_TAG_NONE) {
+		/* Filter by sectag if requested. */
+		if (sectag_filter != TLS_SEC_TAG_NONE && sectag != sectag_filter) {
+			continue;
+		}
+
+		cred = NULL;
+		/* Scan through all credentials within each sectag */
+		while ((cred = credential_next_get(sectag, cred)) != NULL) {
+			/* Filter by credential type if requested. */
+			if (type_filter != TLS_CREDENTIAL_NONE && cred->type != type_filter) {
+				continue;
+			}
+			count++;
+
+			/* Generate a digest of the credential */
+			memset(cred_digest_buf, 0, sizeof(cred_digest_buf));
+			strcpy(cred_digest_buf, "N/A");
+			digest_size = sizeof(cred_digest_buf);
+			err = credential_digest(cred, cred_digest_buf, &digest_size);
+
+			/* Print digest and sectag/type info */
+			shell_fprintf(
+				sh, err ? SHELL_ERROR : SHELL_NORMAL,
+				"%d,%s,%s,%d\n",
+				sectag, cred_type_name_compact(cred->type),
+				err ? "ERROR" : cred_digest_buf, err
+			);
+
+			err = 0;
+		}
+	};
+
+	shell_fprintf(
+		sh, SHELL_NORMAL,
+		"%d credentials found. \n", count
+	);
+
+cleanup:
+	/* Clear digest buffer afterwards for good measure. */
+	memset(cred_digest_buf, 0, sizeof(cred_digest_buf));
+
+	credentials_unlock();
+	return 0;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(tls_cred_cmds,
+	SHELL_CMD_ARG(buf, NULL,
+				"Buffer in credential data so it can be added.",
+				tls_cred_cmd_buf, 2, 0),
+	SHELL_CMD_ARG(add, NULL,
+				"Add a TLS credential.",
+				tls_cred_cmd_add, 5, 1),
+	SHELL_CMD_ARG(del, NULL,
+				"Delete a TLS credential.",
+				tls_cred_cmd_del, 3, 0),
+	SHELL_CMD_ARG(get, NULL,
+				"Retrieve the contents of a TLS credential",
+				tls_cred_cmd_get, 4, 0),
+	SHELL_CMD_ARG(list, NULL,
+				"List stored TLS credentials, optionally filtering by type "
+				"or sectag.",
+				tls_cred_cmd_list, 1, 2),
```

Comment:
Something's off with the indentations here?

Reply:
Sorry, left-over artifact from when I had much longer shell descriptions.

I will soon push corrected indentation, though there's a chance you will still prefer different formatting than what I changed to, so don't hesitate to let me know if that's the case